segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/StorageFactoryCreator.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.host;
+
+import io.pravega.common.util.ServiceBuilderConfig;
+import io.pravega.segmentstore.storage.StorageFactory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Utility class to create a StorageFactory from given configuronion.
+ */
+public final class StorageFactoryCreator {
+    public static StorageFactory createStorageFactoryFromClassName(ServiceBuilderConfig config, String storageImplementation, ScheduledExecutorService executor)
```

Comment:
Probably makes sense for the classname to be the first parameter. I would recommend a better name like `factoryClassName` as the name of the parameter.